crates/rpc/rpc/src/eth/api/traits/transaction.rs

```diff
@@ -56,11 +68,35 @@ pub type StateCacheDB = CacheDB<StateProviderDatabase<StateProviderBox>>;
 /// This implementation follows the behaviour of Geth and disables the basefee check for tracing.
 #[async_trait::async_trait]
 pub trait EthTransactions: Send + Sync {
+    /// Transaction pool with pending transactions. [`TransactionPool::Transaction`] is the
+    /// supported transaction type.
+    type Pool: TransactionPool;
```

Comment:
we need this for

* tx count
* pending block
?

Reply:
there is a function that uses the associated type on `TransactionPool` trait, so couldn't use same pattern like other traits of having a method with return type `&impl TransactionPool`, because then can't ref that type